lib/ansible/utils/module_docs_fragments/cisco_ucs.py

```diff
@@ -0,0 +1,69 @@
+# This code is part of Ansible, but is an independent component.
+# This particular file snippet, and this file snippet only, is BSD licensed.
+# Modules you write using this snippet, which is embedded dynamically by Ansible
+# still belong to the author of the module, and may assign their own license
+# to the complete work.
+#
+# (c) 2016 Red Hat Inc.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright notice,
+#      this list of conditions and the following disclaimer in the documentation
+#      and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+
+class ModuleDocFragment(object):
+
+    # Cisco UCS doc fragment
+    DOCUMENTATION = '''
+
+options:
+    ucs_ip:
+        description:
+            - IP address or hostname of the Cisco UCS server.
+        type: str
+    ucs_username:
+        description:
+            - Username as configured on Cisco UCS server.
+        type: str
+        default: admin
+    ucs_password:
+        description:
+            - Password as configured on Cisco UCS server.
+        type: str
+    port:
+        description:
+            - Port number to be used during connection.(By default uses 443 for https and 80 for http connection)
+        type: int
+    secure:
+        description:
+            - True for secure connection, otherwise False.
+        type: bool
+    proxy:
```

Comment:
So by default Ansible modules use the system-configured proxy settings, unless you set **no_proxy** to **yes**.

Reply:
By default `proxy=None`.  This implies module will use environment proxy.
If any value is assigned to `proxy` parameter , module will use the respective value.
Note: `proxy` is not bool parameter.

Reply:
No, obviously **proxy** is not a boolean, but **no_proxy** is.

So the standard behavior of Ansible modules is to use the system's configuration (**http_proxy**, **https_proxy**) but in order to disable this default behavior, there is the **no_proxy** option, that defaults to **no**. And if you want to use a different proxy, you can set the **proxy** parameter.

In you case there is no way to tell the module *not* to use the system-configured proxy server.

Reply:
`proxy=None` - Uses system defined Proxy
`proxy={}` - Not to use system defined Proxy.
`proxy=http:\\xx.xx.xx.xx:port` - Use this specific Proxy

Reply:
@ragupta-git So I feel I have to repeat myself over and over again. Please use the **use_proxy** variable for disabling proxy support. This is how most (if not all) other modules are doing it.